kumascript/macros/CSSSyntax.ejs

```
@@ -1,442 +1,324 @@
+    if (newValues) {
+      newSyntaxes += ` | ${newValues}`;
     }
-
-    return "<a href=\"" + link + "\" title=\"" + title + "\">&lt;" + type + "&gt;</a>";
+  }
+  return syntax;
 }
 
-function addBrackets(str) {
-    return str.match(/\(\)$/) ? str : str + "()";
+/**
+ * Determines the markup to generate for a single node in the AST
+ * generated by css-tree.
+ */
+function renderNode(name, node) {
+  switch (node.type) {
+    case 'Property': {
+      return `<span class="token property">${name}</span>`;
+    }
+    case 'Type': {
+      // encode < and >
+      let encoded = replaceAll(name, '<', '&lt;');
+      encoded = replaceAll(encoded, '>', '&gt;');
+      // add CSS class: we use "property" because there isn't one for types
+      const span = `<span class="token property">${encoded}</span>`;
+      // if this type is not included in the syntax, link to its dedicated page
+      if (valuespaces[name] && valuespaces[name].value) {
+        return span;
+      } else {
+        // the slug does not include the angle brackets
+        let slug = replaceAll(name, '<', '');
+        slug = replaceAll(slug, '>', '');
+        slug = slug.replace(/\[.*\]/, '')
+        return `<a href="/${locale}/docs/Web/CSS/${slug}">${span}</a>`;
+      }
+    }
+    case 'Multiplier': {
+      // link to the Value Definition Symtax and provide a tooltip
+      let key = name;
+      if (name.startsWith('{')) {
+        key = '{}';
+      }
+      const info = syntaxDescriptions[key];
+      return `<a href="${valueDefinitionUrl}#${info.fragment}" title="${info.tooltip}">${name}</a>`;
+    }
+    case 'Keyword': {
+      return `<span class="token keyword">${name}</span>`;
+    }
+    case 'Function': {
+      return `<span class="token function">${name}</span>`;
+    }
+    case 'Token': {
+      if (name === ')') {
+        // this is probably a closing bracket
+        return `<span class="token function">${name}</span>`;
+      }
```

What if it's _not_ `)`? It would fall-through to 'Group', is that intended?

---

I can't imagine what else it could be, and I have tried this out on a bunch of pages and haven't found any cases where it isn't, so have removed "probably" :). Perhaps I should throw an error if it isn't?